### PR TITLE
Sequelize typescript

### PR DESCRIPTION
**Pull request by ArturFedorov**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -47,9 +47,11 @@
     "pg-hstore": "^2.3.2",
     "pptxgenjs": "^1.10.0",
     "primeng": "^4.2.1",
+    "reflect-metadata": "^0.1.12",
```
Comment (Igogrek) [inline package.json]: move to dev dependencies

File: server/features/employee/employee.service.ts [hunk 2]
```diff
@@ -1,14 +1,12 @@
...
 const employeeService = {
   // GET list of projects with teamcount
-  getEmployees: () => models.Employee.findAll({
-    include: [{
-      as: 'schedules',
-      model: models.Schedule
-    }]
-  }),
-  getRoles: async () => await models.Role.findAll({
+  getEmployees: () => Employee.scope(Scopes.WITH_SCHEDULES).findAll({}),
```
Comment (Igogrek) [inline server/features/employee/employee.service.ts]: {} needed at all?

File: server/features/project/project.service.ts [hunk 3]
```diff
@@ -1,80 +1,50 @@
-import * as models from '../../models/index';
+import {Project} from '../../models/Project';
+import {Scopes} from '../../sequelize/Scopes';
+import {Technology} from '../../models/Technology';
+import { Schedule } from '../../models/Schedule';
```
Comment (Igogrek) [inline server/features/project/project.service.ts]: Formatting

File: server/features/project/project.service.ts [hunk 4]
```diff
@@ -100,26 +70,26 @@ const projectService = {
...
+      const instances = technologies.map(tech => Technology.build(tech));
+      projectNew.$set('technologies', instances);
 
-      const technologies = parse.parseTechnology(projectNew.technologies);
-      const instances = technologies.map(tech => models.Technology.build(tech));
-      projectNew.setTechnologies(instances);
-
-      const schedules = parse.parseShedules(projectNew, projectNew.schedules);
-      models.Schedule.bulkCreate(schedules);
+      // const schedules = parse.parseShedules(projectNew, projectNew.schedules);
```
Comment (Igogrek) [inline server/features/project/project.service.ts]: Either remove commented code or add comment

File: server/sequelize/sequelize.config.ts [hunk 5]
```diff
@@ -0,0 +1,11 @@
+export const Configuration = {
+    development: {
```
Comment (Igogrek) [inline server/sequelize/sequelize.config.ts]: Add test config as it is in develop. Remove old config

File: server/sequelize/sequelize.ts [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import {Sequelize} from 'sequelize-typescript';
+import {Configuration} from './sequelize.config';
+
+export const sequelize = new Sequelize(Configuration['development']);
```
Comment (Igogrek) [inline server/sequelize/sequelize.ts]: Don't force configuration - will not work for CI - hard to maintain

File: server/sequelize/sequelize.config.ts (hunk 5 above)
Comment (ArturFedorov) [reply server/sequelize/sequelize.config.ts]: Should i add production as well as it is in develop

File: server/sequelize/sequelize.ts (hunk 6 above)
Comment (ArturFedorov) [reply server/sequelize/sequelize.ts]: What is the best way to switch between configurations depending on environment? process.env.NODE_ENV || 'development'?

Comment (Igogrek) [reply server/sequelize/sequelize.ts]: Yes. This is ok

File: server/sequelize/sequelize.config.ts (hunk 5 above)
Comment (Igogrek) [reply server/sequelize/sequelize.config.ts]: For now - doesn't matter. Do if production in develop is really used

File: server/sequelize/sequelize.config.ts [hunk 7]
```diff
@@ -7,5 +7,21 @@ export const Configuration = {
       port: 3306,
       password: '1Ass-4ole2',
       modelPaths: [`${__dirname}/../models`]
+    },
+    test: {
+      username: 'root',
+      password: 'null',
```
Comment (Igogrek) [inline server/sequelize/sequelize.config.ts]: null as string!!!

**Pull request merged**